nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/security/NiFiRegistrySecurityConfig.java

```diff
@@ -186,7 +185,7 @@ public void handle(final HttpServletRequest request, final HttpServletResponse r
             if (accessDeniedException instanceof CsrfException) {
                 status = HttpServletResponse.SC_FORBIDDEN;
                 message = "Access Denied: CSRF Header and Cookie not matched";
-                logger.info("Access Denied: CSRF Header [{}] not matched: {}", CsrfCookieName.REQUEST_TOKEN.getCookieName(), accessDeniedException.toString());
+                logger.info("Access Denied: CSRF Header [{}] not matched", CsrfCookieName.REQUEST_TOKEN.getCookieName(), accessDeniedException);
```

Comment:
This specific change should be reverted since the stack trace is not necessarily useful for this type of error.

Reply:
@exceptionfactory those five levels are excellent framing.

Levels 2 and 3 are where I'm focused actually.  The user experience will change if we alter this and part of the reason it is commonly like this is users can 'grep' the logs and see the main log line and the heart of the message.  Exceptions are on different lines and there is no guarantee the exception will print in a subsequent line particularly in busy logs.  Now maybe we're less noise now so perhaps this is ok.

But I want to make sure we're considering the best/most appropriate use case.  Logging the same exception over and over is terrible.  But having the core message of the exception available in context with where the error happened/ and was logged is helpful.